modules/perception/onboard/component/lane_detection_component.cc

```diff
@@ -181,30 +189,11 @@ bool LaneDetectionComponent::Init() {
   }
 
   writer_ = node_->CreateWriter<PerceptionLanes>(output_lanes_channel_name_);
-
-  if (InitSensorInfo() != cyber::SUCC) {
-    AERROR << "InitSensorInfo() failed.";
-    return false;
-  }
-  if (InitAlgorithmPlugin() != cyber::SUCC) {
-    AERROR << "InitAlgorithmPlugin() failed.";
-    return false;
-  }
-  if (InitCameraFrames() != cyber::SUCC) {
-    AERROR << "InitCameraFrames() failed.";
-    return false;
-  }
-  if (InitProjectMatrix() != cyber::SUCC) {
-    AERROR << "InitProjectMatrix() failed.";
-    return false;
-  }
-  if (InitMotionService() != cyber::SUCC) {
-    AERROR << "InitMotionService() failed.";
-    return false;
-  }
-  if (InitCameraListeners() != cyber::SUCC) {
-    AERROR << "InitCameraListeners() failed.";
-    return false;
+  for (auto &itor : LaneDetectionComponent::init_func_arry_) {
+    if ((this->*(itor.init_function_))() != cyber::SUCC) {
+      AERROR << itor.init_fun_name_<< "() failed.";
```

Comment:
IMHO, it's not that straight forward for people to understand this piece of code.
Add Perception folks to review.

Reply:
I want to encapsulate these code like as boost::function to adapter our special requirement in future. I want to remove  duplicated code as far as possible

Reply:
I will add some basic tool classes to make it easy to use. This may implement some code or logic to reuse

Reply:
Hi Mulei,  thanks for the suggestion. However, because all the other components kind of follow the old convention, I prefer the keep the old way.